nncf/torch/quantization/quantize_functions.py

```diff
@@ -275,3 +276,62 @@ def decompress_symmetric(input: torch.Tensor, scale: torch.Tensor) -> torch.Tens
     input = input.type(dtype=scale.dtype)
     decompressed_input = input * scale
     return decompressed_input
+
+
+def pack_uint4(tensor: torch.Tensor) -> torch.Tensor:
+    """
+    Packs a tensor containing uint4 values (in the range [0, 15]) into a tensor with uint8 values,
+    where each element stores two uint4 values.
+
+    :param tensor: A tensor of dtype `torch.uint8` where each element represents a uint4 value.
+        The tensor should contain values in the range [0, 15].
+    :return: A packed tensor of dtype `torch.uint8` where each element packs two uint4 values.
+    :raises ValueError: If the input tensor is not of type `torch.uint8`.
+    """
+    if tensor.dtype != torch.uint8:
+        raise ValueError(f"Invalid tensor dtype {tensor.type}. torch.uint8 type is supported.")
+    packed_tensor = tensor.contiguous()
+    packed_tensor = packed_tensor.reshape(-1, 2)
+    packed_tensor = torch.bitwise_and(packed_tensor[..., ::2], 15) | packed_tensor[..., 1::2] << 4
```

Comment:
Some unit tests for 4bit signed/unsigned packing/unpacking would be useful.

Reply:
added test_pack_uint4 and test_pack_int4